modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-values-factory/src/main/java/com/liferay/dynamic/data/mapping/form/values/factory/internal/DDMFormValuesFactoryImpl.java

```diff
@@ -137,18 +144,30 @@ private void _checkDDMFormFieldParameterNames(
 				ddmFormFieldParameterNames.add(
 					defaultDDMFormFieldParameterName);
 
-				_checkDDMFormFieldParameterNames(
-					ddmFormField.getNestedDDMFormFields(), StringPool.BLANK,
-					ddmFormFieldParameterNames);
+				if (!fullDDMFormFields.containsKey(ddmFormField.getName())) {
+					fullDDMFormFields.put(ddmFormField.getName(), ddmFormField);
+
+					_checkDDMFormFieldParameterNames(
+						ddmFormField.getNestedDDMFormFields(), StringPool.BLANK,
+						ddmFormFieldParameterNames, fullDDMFormFields);
+				}
 			}
 
 			for (String filteredDDMFormFieldParameterName :
 					filteredDDMFormFieldParameterNames) {
 
-				_checkDDMFormFieldParameterNames(
-					ddmFormField.getNestedDDMFormFields(),
-					filteredDDMFormFieldParameterName,
-					ddmFormFieldParameterNames);
+				if (!fullDDMFormFields.containsKey(ddmFormField.getName())) {
+					fullDDMFormFields.put(ddmFormField.getName(), ddmFormField);
+
+					_checkDDMFormFieldParameterNames(
+						ddmFormField.getNestedDDMFormFields(),
+						filteredDDMFormFieldParameterName,
+						ddmFormFieldParameterNames, fullDDMFormFields);
+				}
+			}
+
+			if (!fullDDMFormFields.containsKey(ddmFormField.getName())) {
```

Comment:
Same comment about the importance of this condition here, since we have the opposite condition in lines 125-128.